frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -0,0 +1,207 @@
+    <div className="grid-col flex-1 display-flex flex-column flex-align-center margin-top-10">
+        <div className="grid-row">
+            <Spinner />
+        </div>
+        <div className="text-center">{message}</div>
+    </div>
+);
+
+const SendersDisplay = ({ senders }: { senders: RSSender[] }) => {
+    const navigate = useNavigate();
+    const { state, dispatch } = useManagePublicKey();
+    const [fileContent, setFileContent] = useState("");
+    const [selectedSender, setSelectedSender] = useState(
+        senders.length === 1 ? senders[0].name : ""
+    );
+    const [showUploadDisplay, setShowUploadDisplay] = useState(
+        senders.length === 1 ? true : false
+    );
+    const { fileInputResetValue, fileName, localError } = state;
+
+    const handleSenderSubmit = () => {
+        setShowUploadDisplay(true);
+    };
+
+    // TODO: mocked for now - make the call you need when sending the file
+    const { sendFile, isWorking } = {
+        sendFile: ({}) => {
+            return { fileName: "fileName" };
+        },
+        isWorking: false,
+    };
+
+    const resetState = () => {
+        setFileContent("");
+        dispatch({ type: ManagePublicKeyActionType.RESET });
+    };
+
+    const handleFileChange = async (
+        event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+        // No file selected
+        if (!event?.target?.files?.length) return;
+
+        const file = event.target.files.item(0);
+        if (!file) return;
+
+        const content = await file.text();
+
+        setFileContent(content);
+
+        dispatch({
+            type: ManagePublicKeyActionType.FILE_SELECTED,
+            payload: { file },
+        });
+    };
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        if (fileContent.length === 0) {
+            showError("No file contents to validate.");
+            return;
+        }
+
+        // initializes necessary state and sets `isSubmitting`
+        dispatch({ type: ManagePublicKeyActionType.PREPARE_FOR_REQUEST });
+
+        try {
+            sendFile({
+                contentType: ContentType.PEM,
+                fileContent: fileContent,
+                fileName: fileName,
+                selectedSender: selectedSender,
+            });
+        } catch (e: any) {
+            console.trace(e);
+            showError(`Uploading public key failed. ${e.toString()}`);
+        }
+    };
+
+    const handleBack = () => {
+        resetState();
+        navigate("/resources");
+    };
+
+    // TODO: add more logic here to toggle between true/false
+    const submitted = false;
+
+    useEffect(() => {
+        if (localError) {
+            showError(localError);
+        }
+    }, [localError]);
+
+    return (
+        <div className="grid-col-12">
+            {senders?.length > 1 && !showUploadDisplay && (
+                <Form name="senderSelect" onSubmit={() => handleSenderSubmit()}>
```

Comment:
Is using Form overkill here?

Reply:
I think it's the correct use of Form since this is used for data submission :)